src/SFA.DAS.Provider.PR.Web.UnitTests/Models/HomeSubmitModelTests.cs

```diff
@@ -0,0 +1,93 @@
+﻿using FluentAssertions;
+using SFA.DAS.Provider.PR.Web.Models;
+
+namespace SFA.DAS.Provider.PR_Web.UnitTests.Models;
+
+public class HomeSubmitModelTests
+{
+    [Test]
+    public void ToQueryString_SearchTermHasValue_AddsSearchTerm()
+    {
+        HomeSubmitModel sut = new() { SearchTerm = "search term" };
+
+        var actual = sut.ToQueryString();
+
+        actual.Should().ContainKey(nameof(HomeSubmitModel.SearchTerm)).WhoseValue.Should().Be(sut.SearchTerm);
+    }
+
+    [TestCase("")]
+    [TestCase("  ")]
+    public void ToQueryString_SearchTermIsEmptyOrWhiteSpace_DoesNotAddSearchTerm(string searchTerm)
+    {
+        HomeSubmitModel sut = new() { SearchTerm = searchTerm };
+
+        var actual = sut.ToQueryString();
+
+        actual.Should().NotContainKey(nameof(HomeSubmitModel.SearchTerm));
+    }
+
+    [Test]
+    public void ToQueryString_HasPendingRequestIsTrue_AddsToDictionary()
+    {
+        HomeSubmitModel sut = new() { HasPendingRequest = true };
+
+        var actual = sut.ToQueryString();
+
+        actual.Should().ContainKey(nameof(HomeSubmitModel.HasPendingRequest)).WhoseValue.Should().Be(true.ToString());
+    }
+
+    [Test]
+    public void ToQueryString_HasPendingRequestIsFalse_DoesNotAddToDictionary()
+    {
+        HomeSubmitModel sut = new() { HasPendingRequest = false };
+
+        var actual = sut.ToQueryString();
+
+        actual.Should().NotContainKey(nameof(HomeSubmitModel.HasPendingRequest));
+    }
+
+    [TestCase(true, false, true)]
+    [TestCase(false, true, true)]
+    [TestCase(false, false, false)]
+    [TestCase(true, true, false)]
+    public void ToQueryString_ConditionallyAddsHasCreateCohortPermission(bool yesSelected, bool noSelected, bool isAdded)
+    {
+        HomeSubmitModel sut = new() { HasAddApprenticePermission = yesSelected, HasNoAddApprenticePermission = noSelected };
+
+        var actual = sut.ToQueryString();
+
+        if (isAdded)
+        {
+            actual.Should().ContainKey(HomeSubmitModel.HasCreateCohortPermissionKey).WhoseValue.Should().Be(yesSelected.ToString());
+        }
+        else
+        {
+            actual.Should().NotContainKey(HomeSubmitModel.HasCreateCohortPermissionKey);
+        }
+    }
+
+    [TestCase(true, false, false, true)]
+    [TestCase(false, true, false, true)]
+    [TestCase(false, false, true, true)]
+    [TestCase(true, true, true, true)]
```

Comment:
I could be wrong here, but I believe the test case should be 'true','true','true', 'false' ie if they are all true, it should not be added

Reply:
I have gone with if any of them are checked, I pass in all the params, otherwise the code was getting too complicated.  It is easier to not have that logic here, the inner deals with it.

Reply:
ok fair enough